juriscraper/pacer/http.py

```diff
@@ -0,0 +1,170 @@
+import requests
+
+from juriscraper.lib.log_tools import make_default_logger
+
+logger = make_default_logger()
+
+
+class PacerSession(requests.Session):
+    """
+    Extension of requests.Session to handle PACER oddities making it easier
+    for folks to just POST data to PACER endpoints/apis
+    """
+
+    def __init__(self, pacer_token=None, cookie_jar=None):
+        """
+        Instantiate a new PACER HTTP Session with some Juriscraper defaults
+        :param pacer_token: a PACER_SESSION token value
+        """
+        super(PacerSession, self).__init__()
+        self.headers['User-Agent'] = 'Juriscraper'
+        self.verify = False
+
+        if pacer_token:
+            self.cookies.set('PacerSession',
+                             pacer_token,
+                             domain='.uscourts.gov',
+                             path='/')
+
+        if cookie_jar:
+            self.cookies = cookie_jar
+
+    def post(self, url, data=None, json=None, **kwargs):
+        """
+        Overrides requests.Session.post with PACER-specific fun.
+
+        Will automatically convert data dict into proper multi-part form data
+        and pass to the files parameter instead.
+
+        Will set a timeout of 300 if not provided.
+
+        All other uses or parameters will pass through untouched
+        :param url: url string to post to
+        :param data: post data
+        :param json: json object to post
+        :param kwargs: assorted keyword arguments
+        :return: requests.Response
+        """
+        kwargs.setdefault('timeout', 300)
+
+        if data:
+            pacer_data = self._prepare_multipart_form_data(data)
+            return super(PacerSession, self).post(url, files=pacer_data, **kwargs)
+
+        return super(PacerSession, self).post(url, data=data, json=json, **kwargs)
+
+    @staticmethod
+    def _prepare_multipart_form_data(data):
+        """
+        Transforms a data dictionary into the multi-part form data that PACER
+        expects as the POST body
+        :param data: dict of data to transform
+        :return: dict with values wrapped into tuples like:(None, <value>)
+        """
+        output = dict()
+        for key in data:
+            output[key] = (None, data[key])
+        return output
+
+
+def _make_login_url(court_id):
+    """Make a login URL for a given court id."""
+    if court_id == 'psc':
+        # training account
+        return 'https://dcecf.psc.uscourts.gov/cgi-bin/login.pl'
+    return 'https://pacer.login.uscourts.gov/csologin/login.jsf?pscCourtId=%s' % court_id
+
+
+def login(court_id, username, password):
+    """
+    Log into a PACER jurisdiction via the main PACER portal which should set our global PacerSession
+
+    :param court_id: id of the court to authenticate with
+    :param username: PACER username
+    :param password: PACER password
+    :return: new PacerSession configured with PacerSession token in cookie
+    """
+    if username == 'tr1234':
```

Comment:
I thought about that but wasn't sure if the court_id `psc` or whatever was the test id. Is that the case?